client/src/components/ListComponent.tsx

```diff
@@ -11,13 +11,54 @@ type Props = {
   reservation: ReservationData;
   className?: string;
 };
-const dateFormatter = new Intl.DateTimeFormat('ko-KR', { year: 'numeric', month: '2-digit', day: '2-digit' });
+const dateFormatter = new Intl.DateTimeFormat('ko-KR', { year: 'numeric', month: '2-digit', day: '2-digit', hour: '2-digit' });
 
 function ListComponent({ type, reservation, className }: Props) {
   const [buttonText, setButtonText] = useState('');
   const [buttonType, setButtonType] = useState<ButtonType>('disabled');
   const [buttonClick, setButtonClick] = useState<((reservation: ReservationData) => void) | null>(null);
   const [rentStatus, setRentStatus] = useState(reservation.rentStatus);
+  const [timeDifference, setTimeDifference] = useState(0);
+  const [extraFee, setExtraFee] = useState(0);
+
+  useEffect(() => {
+    const updateTimeDifference = () => {
+      const returnTime = new Date(Number(reservation.rentPeriod[1]));
+      // 현재 시간과 반납 시간의 차이 계산 전, 숫자 타입 확인
+      const now = new Date().getTime(); // getTime()은 항상 숫자를 반환
+      const returnTimeMs = returnTime.getTime(); // 반납 시간을 밀리초로 변환
+
+      // 이제 returnTimeMs와 now는 숫자 타입이므로 산술 연산 가능
+      const difference = returnTimeMs - now;
+
+      if (difference < 0) {
+        const hours = Math.floor(difference / (1000 * 60 * 60));
+        setTimeDifference(hours * -1);
+      }
+    };
```

Comment:
```suggestion
      // 이제 returnTimeMs와 now는 숫자 타입이므로 산술 연산 가능
      const difference = now - returnTimeMs;

      if (difference > 0) {
        const hours = Math.floor(difference / (1000 * 60 * 60));
        setTimeDifference(hours);
      }
    };
```
이러면 가독성이 더 좋아질 것 같아요!

Reply:
반영하겠습니다